src/main/scala/com/github/takezoe/solr/scala/QueryBuilderBase.scala

```diff
@@ -266,7 +266,17 @@ trait QueryBuilderBase[Repr <: QueryBuilderBase[Repr]] {
           field.getValues.asScala.map { value => (value.getName, value.getCount) }.toMap
       )}.toMap
     }
-    MapQueryResult(numFound,numGroupsFound, queryResult, groupResult, facetResult,response.getQTime)
+    val facetPivotResult = response.getFacetPivot match {
+      case null => Map.empty[String, List[FacetPivot]]
+      case facetPivots =>
+        facetPivots.asScala.map { field =>
+          (
+            field.getKey,
+            field.getValue.asInstanceOf[List[FacetPivot]]
```

Comment:
Does this work? I think just cast `Object` to `List[FacetPivot]` doesn't work. An explicit conversion would be necessary.

Reply:
Hey, here getValue is nothing but List<PivotField>. So we are basically trying to convert List<PivotField> to List<FacetPivot>.

https://lucene.apache.org/solr/7_0_1/solr-solrj/org/apache/solr/client/solrj/response/QueryResponse.html#getFacetPivot--

I also tried verifying by following code snippet:
```
object test {
 def main(args: Array[String]) : Unit ={
   val tempList = new java.util.ArrayList[PivotField];
   val tempListObj = new PivotField("field1", 7, 8, null,
     null, null, null);
   tempList.add(tempListObj);
   val temp1 = new PivotField("field", 5, 6, tempList,
     null, null, null);
   val temp = new java.util.ArrayList[PivotField];
   temp.add(temp1);
   val newTemp = temp.toArray.toList
   val finalresult = newTemp.asInstanceOf[List[FacetPivot]];
   println(finalresult);
 }
}
```

Output:
List(field:5 [6] [field1:7 [8] null])

Let me know your thoughts.!

Reply:
Java's List cannot be cast to Scala's List directly. Also, the list element is SolrJ's FacetPivot. It can't be cast to Scala's FacetPivot which is defined as a case class. This kind of explicit conversion is necessary:
https://github.com/takezoe/solr-scala-client/blob/525b0eca1e06eeedffffc8594fbc31cc9d00908e/src/main/scala/com/github/takezoe/solr/scala/QueryBuilderBase.scala#L249-L255